extensions/http/src/main/java/org/mule/extension/http/internal/StaticResourceLoader.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.http.internal;
+
+import static java.lang.String.format;
+import static org.mule.runtime.core.config.i18n.MessageFactory.createStaticMessage;
+import static org.mule.runtime.core.util.Preconditions.checkArgument;
+import org.mule.extension.http.api.HttpRequestAttributes;
+import org.mule.runtime.api.metadata.MediaType;
+import org.mule.runtime.core.api.MuleEvent;
+import org.mule.runtime.core.api.MuleMessage;
+import org.mule.runtime.core.util.IOUtils;
+import org.mule.runtime.core.util.StringUtils;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.module.http.internal.component.ResourceNotFoundException;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.activation.MimetypesFileTypeMap;
+
+public class StaticResourceLoader
+{
+    private static final String ANY_PATH = "/*";
+    private static final String ROOT_PATH = "/";
+    private static final String DEFAULT_MIME_TYPE = "application/octet-stream";
+    private MimetypesFileTypeMap mimeTypes = new MimetypesFileTypeMap();
+
+    /**
+     * The resource base from where documents are served up. For example: /Users/maxthemule/resources
+     */
+    @Parameter
+    private String resourceBasePath;
+
+    /**
+     * The default file to serve when a directory is specified. The default value is 'index.html'.
+     */
+    @Parameter
+    @Optional(defaultValue = "index.html")
+    private String defaultFile;
+
+    public MuleMessage load(MuleEvent event) throws ResourceNotFoundException
+    {
+        checkArgument(event.getMessage().getAttributes() instanceof HttpRequestAttributes, "Message attributes must be HttpRequestAttributes.");
```

Comment:
Create an issue to track this as something TODO before releasing Mule 4. I would prefer to avoid this kind of requirement from components. As we discussed yesterday, it's better if the tooling helps the user to autocomplete the required input parameters of the load-static-resource operation instead of automatically discovering those from an HttpRequestAttributes object.